frontend/src/metabase/collections/components/CollectionHeader/CollectionCaption.tsx

```diff
@@ -62,15 +62,18 @@ export const CollectionCaption = ({
       </CaptionTitleContainer>
       {(isEditable || hasDescription) && (
         <CaptionDescription
-          key={collection.id}
-          initialValue={collection.description}
+          key={
+            // Including the description in the key prevents a stale value from
+            // being stored in the state of EditableText if the collection's
+            // description is modified in another component
+            `${collection.id}-${collection.description}`
+          }
```

Comment:
This fixes a bug where deleting the description from the sidesheet would cause it to reappear in the under-title description on mouseover